src/main/java/com/project/store/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.project.store.service.impl;
+
+
+import com.project.store.exceptions.messages.ExceptionMessages;
+import com.project.store.exceptions.product.ProductNotFoundException;
+import com.project.store.model.category.Category;
+import com.project.store.model.product.Product;
+import com.project.store.repository.ProductRepository;
+import com.project.store.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+
+    private ProductRepository productRepository;
+
+    private CategoryServiceImpl categoryService;
+
+    @Autowired
```

Comment:
No need to annotate with @autowired if do injection through  constructor

Reply:
Removed @Autowired annotations from service classes